bundles/org.openhab.binding.entsoe/src/main/java/org/openhab/binding/entsoe/internal/EntsoeHandler.java

```diff
@@ -0,0 +1,280 @@
+            Instant nowPlusNext = now.plus(duration);
+            Instant nowMinusNext = now.minus(duration);
+
+            if (getNext && thisEntry.isAfter(now)) {
+                return entry.getValue();
+            }
+            if (!getNext && thisEntry.isAfter(nowMinusNext) && thisEntry.isBefore(nowPlusNext)) {
+                return entry.getValue();
+            }
+        }
+        throw new EntsoeResponseMapException("Could not get actual ENTSOE TimeSerie");
+    }
+
+    private boolean needToFetchHistoricDays() {
+        return needToFetchHistoricDays(false);
+    }
+
+    private boolean needToFetchHistoricDays(boolean updateHistoricDaysInitially) {
+        boolean needToFetch = false;
+        if (historicDaysInitially < config.historicDays) {
+            logger.debug("Need to fetch historic data. Historicdays was changed to a greater number: {}",
+                    config.historicDays);
+            needToFetch = true;
+        }
+
+        if (updateHistoricDaysInitially && historicDaysInitially != config.historicDays) {
+            historicDaysInitially = config.historicDays;
+        }
+
+        return needToFetch;
+    }
+
+    private void refreshPrices() {
+        if (!isLinked(EntsoeBindingConstants.CHANNEL_SPOT_PRICE)) {
+            logger.debug("Channel {} is not linked, skipping channel update",
+                    EntsoeBindingConstants.CHANNEL_SPOT_PRICE);
+            return;
+        }
+
+        config = getConfigAs(EntsoeConfiguration.class);
+
+        if (entsoeTimeSeries.isEmpty()) {
+            fetchNewPrices();
+            return;
+        }
+
+        boolean needsInitialUpdate = lastDayAheadReceived.equals(ZonedDateTime.of(LocalDateTime.MIN, cetZoneId))
+                || needToFetchHistoricDays(true);
+        boolean hasNextDayValue = false;
+        for (Map.Entry<Instant, Optional<EntsoeTimeSerie>> entry : entsoeTimeSeries.entrySet()) {
+            if (entry.getKey().atZone(cetZoneId)
+                    .isAfter(currentCetTimeWholeHours().withHour(config.spotPricesAvailableCetHour))) {
+                hasNextDayValue = true;
+                break;
+            }
+        }
+        boolean readyForNextDayValue = currentCetTime()
+                .isAfter(currentCetTimeWholeHours().withHour(config.spotPricesAvailableCetHour));
+
+        if (needsInitialUpdate || (!hasNextDayValue && readyForNextDayValue)) {
+            fetchNewPrices();
+        } else {
+            updateCurrentState(EntsoeBindingConstants.CHANNEL_SPOT_PRICE);
+            schedule(true);
+        }
+    }
+
+    private void fetchNewPrices() {
+        logger.trace("Fetching new prices");
+
+        ZonedDateTime startUtc = currentCetTimeWholeHours()
+                .minusDays(needToFetchHistoricDays() ? config.historicDays - 1 : 0).withHour(00)
+                .withZoneSameInstant(utcZoneId);
+        ZonedDateTime endUtc = currentCetTimeWholeHours().plusDays(2).withHour(00).withZoneSameInstant(utcZoneId);
```

Comment:
While there is nothing wrong with this, maybe it could be made a bit clearer by using `LocalTime.MIDNIGHT` and skipping the unneeded `truncatedTo(ChronoUnit.HOURS)`?
```suggestion
        ZonedDateTime endUtc = ZonedDateTime.now(cetZoneId).plusDays(2).with(LocalTime.MIDNIGHT).withZoneSameInstant(utcZoneId);
```

or if using Instant for creating Request instance (see other comment):
```suggestion
        Instant endUtc = ZonedDateTime.now(cetZoneId).plusDays(2).with(LocalTime.MIDNIGHT).toInstant();
```